app/src/main/java/com/framgia/moviedbtraining/movies/NowPlayingPresenter.kt

```diff
@@ -0,0 +1,73 @@
+package com.framgia.moviedbtraining.movies
+
+import com.framgia.moviedbtraining.App
+import com.framgia.moviedbtraining.R
+import com.framgia.moviedbtraining.constants.Constants
+import com.framgia.moviedbtraining.model.MoviesResponse
+import com.framgia.moviedbtraining.rest.ApiClient
+import com.framgia.moviedbtraining.rest.ApiInterface
+import com.framgia.moviedbtraining.utils.GeneralUtil
+import com.framgia.moviedbtraining.widget.EndlessRecyclerOnScrollListener
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class NowPlayingPresenter(
+    private val mViewModel: NowPlayingContractNew.ViewModel) : NowPlayingContractNew.Presenter {
+  override fun addEndlessListener() {
+    addEndlessRecyclerOnScrollListener()
+    mViewModel.addEndlessListener(onLoadMoreListener!!)
+  }
+
+  var mPage: Int = 1
+  var isEmpty: Boolean = false
+  private var onLoadMoreListener: EndlessRecyclerOnScrollListener? = null
+
+  override fun getMovies() {
+    getNowPlayingMovies(mPage)
+  }
+
+  private fun getNowPlayingMovies(page: Int) {
+    if ((!GeneralUtil.isNetworkAvailable(App.self()))) {
+      mViewModel.hideLoading()
+      mViewModel.showToast(App.self().getString(R.string.err_network))
+      return
+    }
+    mViewModel.showLoading()
+    val apiService = ApiClient.client.create(ApiInterface::class.java)
```

Comment:
should create a request helper or api helper to hold all service connnection like this